src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,35 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter PATTERN = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder strB = new StringBuilder();
+        if ((names != null) && (data != null)) {
+            LocalDate dateLocalFrom = LocalDate.parse(dateFrom, PATTERN);
+            LocalDate dateLocalTo = LocalDate.parse(dateTo, PATTERN);
+            int[] salaries = new int[names.length];
+            for (String line : data) {
+                String[] strings = line.split(" ");
```

Comment:
All magic numbers, strings, and symbols in your code should be constants.
```suggestion
                String[] strings = line.split(SEPARATOR);
```